lib/custom/tests/MShop/Service/Provider/Payment/DatatransQueryTest.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/**
+ * @license LGPLv3, http://opensource.org/licenses/LGPL-3.0
+ * @copyright Aimeos (aimeos.org), 2019
+ */
+namespace Aimeos\MShop\Service\Provider\Payment;
+
+class DatatransQueryTest extends \PHPUnit\Framework\TestCase
+{
+	private $object;
+	private $context;
+	private $serviceItem;
+
+	protected function setUp()
+	{
+		if ( !class_exists( 'Omnipay\Omnipay' ) ) {
+			$this->markTestSkipped( 'Omnipay library not available' );
+		}
+
+		$this->context = \TestHelper::getContext();
+
+		$serviceManager = \Aimeos\MShop\Service\Manager\Factory::create($this->context);
+		$this->serviceItem = $serviceManager->createItem();
+		$this->serviceItem->setConfig([ 'type' => 'Dummy' ]);
+		$this->serviceItem->setCode('OGONE');
+
+		$methods = [
+			'getCustomerData', 'getOrder', 'getOrderBase', 'getTransactionReference', 'isImplemented',
+			'saveOrder', 'saveOrderBase', 'getXmlProvider', 'saveTransationRef', 'setCustomerData',
+			'getProvider', 'getTransaction',
+		];
+
+		$this->object = $this->getMockBuilder(\Aimeos\MShop\Service\Provider\Payment\Datatrans::class)
+							 ->setConstructorArgs([ $this->context, $this->serviceItem ])
+							 ->setMethods($methods)
+							 ->getMock();
+	}
+
+	protected function tearDown()
+	{
+		unset($this->object, $this->serviceItem, $this->context);
+	}
+
+	public function testDatatransReceivedCode()
+	{
+		$this->assertEquals(self::getObjectAttribute($this->object, 'datatransReceivedCode'), [2, 3, 21]);
+	}
+	public function testDatatransAuthorizedCode()
+	{
+		$this->assertEquals(self::getObjectAttribute($this->object, 'datatransAuthorizedCode'), [1]);
+	}
+
+	public function testTransactionId()
```

Comment:
You only need to test methods and code paths of the code you've written in the Datatrans class. This test method tests if your test code works